numpydoc/validate.py

```diff
@@ -295,7 +295,17 @@ def source_file_def_line(self):
         Number of line where the object is defined in its file.
         """
         try:
-            return inspect.getsourcelines(self.code_obj)[-1]
+            sourcelines = inspect.getsourcelines(self.code_obj)
+            # getsourcelines will return the line of the first decorator found for the
+            # current function. We have to find the def declaration after that.
+            def_lines = [
+                i
+                for i, x in enumerate(
+                    [re.match("^ *(def|class)", s) for s in sourcelines[0]]
```

Comment:
I would make this tiny tweak but I doubt it matters in practice
```suggestion
                    [re.match("^ *(def|class) ", s) for s in sourcelines[0]]
```